core/sequencer/server.go

```diff
@@ -343,21 +341,22 @@ func (s *Server) readMessages(ctx context.Context, source *spb.MapMetadata_Sourc
 	ss := metadata.Source(source)
 	low := ss.StartTime()
 	high := ss.EndTime()
-	// Loop until less than chunkSize items are returned.
-	for count := chunkSize; count == chunkSize; {
-		batch, err := s.logs.ReadLog(ctx, directoryID, source.LogId, low, high, chunkSize)
+	for moreToRead := true; moreToRead; {
+		// Request one more item than chunksize so we can find the next page token
+		batch, err := s.logs.ReadLog(ctx, directoryID, source.LogId, low, high, chunkSize+1)
```

Comment:
Can you double check `ReadLog`'s inclusion/exclusion semantics? The current comment says `(low, high]`, which means you don't need to read one extra item to figure out the next `low`, just set it to be the last ID. Moreover, setting `low` to `chunkSize+1`th ID makes the next `ReadLog` to skip it.

Reply:
The comment was incorrect. It should be `[low, high)`